source/Core/Threads/OperatingModes/DebugMenu.cpp

```diff
@@ -4,6 +4,39 @@ extern osThreadId    MOVTaskHandle;
 extern osThreadId    PIDTaskHandle;
 extern OperatingMode currentMode;
 
+int8_t getPowerSourceNumber(void) {
```

Comment:
Probably worth moving this into its own file with matching header, since its no longer coupled to debug menu

Reply:
> its no longer coupled to debug menu

Technically it still is but just reused in another place at the same time. Plus, I always thought that the less **new files** in a patch the better but it depends on a project I guess. So, to be clear though - are you sure that this demands creating a separate translation unit? Just asking, Captn. If so, then tell me the name of a new file and the location in the source tree - it's not mine ship after all ;)

Reply:
This could just be a `powerSource.c` file in the same folder; I view files more as wanting to be able to know what file to open when I first open the repo. So I dont like overly deep mess'es of files with similar names. But also I dont like burying a shared function in a file that sounds like it only does one thing.

To me the question would be "given the function getPowerSourceNumber; which of these files has it in" and thats not clear.